presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotFilterExpressionConverter.java

```diff
@@ -141,6 +187,21 @@ private PinotExpression handleCast(CallExpression cast, Function<VariableReferen
             if (typeManager.canCoerce(input.getType(), expectedType)) {
                 return input.accept(this, context);
             }
+            if (expectedType == DateType.DATE) {
+                try {
+                    PinotExpression expression = input.accept(this, context);
+                    if (input.getType() == TimestampType.TIMESTAMP || input.getType() == TimestampWithTimeZoneType.TIMESTAMP_WITH_TIME_ZONE) {
+                        return expression;
+                    }
+                    if (input.getType() == VarcharType.VARCHAR) {
+                        Integer daysSinceEpoch = (int) TimeUnit.MILLISECONDS.toDays(DATE_FORMATTER.parseMillis(expression.getDefinition().substring(1, expression.getDefinition().length() - 1)));
```

Comment:
Perhaps a comment here about why the substring offsets are chosen ?

Is this branch about handling of literals / constants ? Is there another way to check for a RowExpression being a constant/literal rather than checking for its type ?

Reply:
this substring is to remove the leading and tailing quotes